examples/abitest/module_0/rust/src/proto/abitest_grpc.rs

```diff
@@ -24,16 +24,28 @@ use protobuf::Message;
 use std::io::Write;
 
 // Oak Node server interface
-pub trait OakABITestServiceNode {
+pub trait OakABITestService {
     fn run_tests(&mut self, req: super::abitest::ABITestRequest) -> grpc::Result<super::abitest::ABITestResponse>;
 }
 
 // Oak Node gRPC method dispatcher
-pub fn dispatch<T: OakABITestServiceNode>(node: &mut T, method: &str, req: &[u8], writer: grpc::ChannelResponseWriter) {
+pub fn dispatch<T: OakABITestService>(node: &mut T, method: &str, req: &[u8], writer: grpc::ChannelResponseWriter) {
     match method {
         "/oak.examples.abitest.OakABITestService/RunTests" => grpc::handle_req_rsp(|r| node.run_tests(r), req, writer),
         _ => {
             panic!("unknown method name: {}", method);
         }
     };
 }
+
+// Client interface trait
+pub trait OakABITestServiceClient {
+    fn run_tests(&self, req: super::abitest::ABITestRequest) -> grpc::Result<super::abitest::ABITestResponse>;
+}
+
+// Client trait implementation
+impl OakABITestServiceClient for oak::grpc::client::Client {
+    fn run_tests(&self, req: super::abitest::ABITestRequest) -> grpc::Result<super::abitest::ABITestResponse> {
+        oak::grpc::invoke_grpc_method("/oak.examples.abitest.OakABITestService/RunTests", req, &self.invocation_sender)
+    }
+}
```

Comment:
I find it a bit awkward that multiple client traits may be implemented for the same `Client` instance: by passing around a `Client` instance by itself we cannot enforce the type of the actual client it represents.

How about instead generating a struct that wraps the `Client` instance, and implement the client trait on that?

Reply:
Sounds good; it doesn't even need to be a `trait` then.